src/refract/Element.cc

```diff
@@ -129,4 +131,243 @@ namespace refract
+                    for (IElement::MemberElementCollection::const_iterator it = append.begin() ; it != append.end() ; ++it) {
+                        if (StringElement* key = TypeQueryVisitor::as<StringElement>((*it)->value.first)) {
+                            IElement::MemberElementCollection::iterator item = info.find(key->value);
+
+                            if (item != info.end()) {
+                                delete (*item)->value.second;
+                                (*item)->value.second = (*it)->value.second->clone();
+                            }
+                        }
+                    }
+                }
+            };
+
+            /**
+             * precondition - target && append element MUST BE of same type
+             * we use static_cast<> without checking type this is responsibility of caller
+             */
+            template <typename T>
+            static void doMerge(IElement* target, const IElement* append) {
+                typedef T ElementType;
+
+                InfoMerge<T>(target->meta)(append->meta);
+                InfoMerge<T>(target->attributes)(append->attributes);
+
+                ValueMerge<T>(static_cast<ElementType&>(*target))
+                             (static_cast<const ElementType&>(*append));
+            }
+
+        public:
+
+            ElementMerger() : result(NULL), base(TypeQueryVisitor::Unknown) {}
+
+            void operator()(const IElement* e) {
+                if (!e) {
+                    return;
+                }
+
+                if (!result) {
+                    result = e->clone();
+
+                    TypeQueryVisitor type;
+                    type.visit(*result);
+                    base = type.get();
+                    return;
+                }
+
+                TypeQueryVisitor type;
+                e->content(type);
+
+                if(type.get() != base) {
+                    throw refract::LogicError("Can not merge different types of elements");
+                }
+
+                switch(base) {
+                    case TypeQueryVisitor::String:
+                        doMerge<StringElement>(result, e);
+                        return;
+
+                    case TypeQueryVisitor::Number:
+                        doMerge<NumberElement>(result, e);
+                        return;
+
+                    case TypeQueryVisitor::Boolean:
+                        doMerge<BooleanElement>(result, e);
+                        return;
+
+                    case TypeQueryVisitor::Array:
+                        doMerge<ArrayElement>(result, e);
+                        return;
+
+                    case TypeQueryVisitor::Object:
+                        doMerge<ObjectElement>(result, e);
+                        return;
+
+                    case TypeQueryVisitor::Member:
+                        doMerge<MemberElement>(result, e);
+                        return;
+
+                    case TypeQueryVisitor::Null:
```

Comment:
shouldn't these two do something, at least throw an error or something? Or put comment why we can do nothing